src/Test/QuickCheck/Instances.hs

```diff
@@ -518,3 +464,66 @@ instance CoArbitrary b => CoArbitrary (Tagged.Tagged a b) where
 
 instance Function b => Function (Tagged.Tagged a b) where
     function = functionMap Tagged.unTagged Tagged.Tagged
+
+instance Arbitrary1 Proxy where
+  liftArbitrary _ = pure Proxy
+  liftShrink _ _ = []
+
+instance Arbitrary (Proxy a) where
+  arbitrary = pure Proxy
+  shrink _  = []
+
+instance CoArbitrary (Proxy a) where
+  coarbitrary _ = id
+
+instance Function (Proxy a) where
+  function = functionMap (const ()) (const Proxy)
```

Comment:
Are these instance guarded by `CPP` already, so that they are only included if `QuickCheck >= 2.10`?  If not, that might smoothen the upgrade experience.